core/src/test/java/uk/gov/justice/services/core/handler/HandlerInstanceAndMethodTest.java

```diff
@@ -29,44 +27,48 @@
-        assertThat(handlerInstanceAndMethod, notNullValue());
-
-        handlerInstanceAndMethod.execute(envelope);
+        handlerInstanceWithMethod().execute(envelope);
         verify(commandHandler).handler1(envelope);
     }
 
     @Test(expected = HandlerExecutionException.class)
-    public void shouldThrowExceptionWithExceptionFromuteThrowsException() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException, IOException {
+    public void shouldThrowHandlerExecutionExceptionIfExceptionThrown() throws Exception {
         Envelope envelope = testEnvelope("envelope.json");
-        List<Method> methods = HandlerUtil.findHandlerMethods(CommandHandler.class, Handles.class);
-        HandlerInstanceAndMethod handlerInstanceAndMethod = new HandlerInstanceAndMethod(commandHandler, methods.get(0));
-        assertThat(handlerInstanceAndMethod, CoreMatchers.notNullValue());
-
         doThrow(new RuntimeException()).when(commandHandler).handler1(envelope);
-        handlerInstanceAndMethod.execute(envelope);
+        handlerInstanceWithMethod().execute(envelope);
     }
 
     @Test
     public void shouldReturnStringDescriptionOfHandlerInstanceAndMethod() {
-        List<Method> methods = HandlerUtil.findHandlerMethods(CommandHandler.class, Handles.class);
-        HandlerInstanceAndMethod handlerInstanceAndMethod = new HandlerInstanceAndMethod(commandHandler, methods.get(0));
-        assertThat(handlerInstanceAndMethod.toString(), CoreMatchers.notNullValue());
+        assertThat(handlerInstanceWithMethod().toString(), notNullValue());
     }
 
     @Test
     public void shouldNotReturnNullWithNullHandlerInstance() {
-        List<Method> methods = HandlerUtil.findHandlerMethods(CommandHandler.class, Handles.class);
-        HandlerInstanceAndMethod handlerInstanceAndMethod = new HandlerInstanceAndMethod(null, methods.get(0));
-        assertThat(handlerInstanceAndMethod.toString(), CoreMatchers.notNullValue());
+        assertThat(nullHandlerInstanceWithMethod().toString(), notNullValue());
     }
 
     @Test
     public void shouldNotReturnNullWithNullMethod() {
-        assertThat(new HandlerInstanceAndMethod(commandHandler, null).toString(), CoreMatchers.notNullValue());
+        assertThat(handlerInstanceWithNullMethod().toString(), notNullValue());
+    }
+
+    private HandlerInstanceAndMethod handlerInstanceWithNullMethod() {
+        return new HandlerInstanceAndMethod(commandHandler, null);
+    }
+
+    private HandlerInstanceAndMethod nullHandlerInstanceWithMethod() {
+        return new HandlerInstanceAndMethod(null, methods().get(0));
+    }
+
+    private HandlerInstanceAndMethod handlerInstanceWithMethod() {
+        return new HandlerInstanceAndMethod(commandHandler, methods().get(0));
+    }
+
+    private List<Method> methods() {
+        return HandlerUtil.findHandlerMethods(CommandHandler.class, Handles.class);
     }
 
     private Envelope testEnvelope(String fileName) throws IOException {
```

Comment:
I think we said that we would always use braces rather than flattening to one line as SonarQube reports on this.

Is there a better name than handler1 for the method? - make it clearer.